tabpfn_client/prompt_agent.py

```diff
@@ -23,45 +23,55 @@ def prompt_welcome(cls):
 
     @classmethod
     def prompt_and_set_token(cls, user_auth_handler: "UserAuthenticationClient"):
+        # Choose between registration and login
         prompt = "\n".join([
             "Please choose one of the following options:",
             "(1) Create a TabPFN account",
             "(2) Login to your TabPFN account",
             "",
             "Please enter your choice: ",
         ])
+        choice = cls._choice_with_retries(prompt, ["1", "2"])
 
-        choice = input(cls.indent(prompt))
-
+        # Registration
         if choice == "1":
             #validation_link = input(cls.indent("Please enter your secret code: "))
             validation_link = "tabpfn-2023"
             # create account
-            email = input(cls.indent("Please enter your email: "))
-
-            password_req = user_auth_handler.get_password_policy()
-            password_req_prompt = "\n".join([
-                "",
-                "Password requirements (minimum):",
-                "\n".join([f". {req}" for req in password_req]),
-                "",
-                "Please enter your password: ",
-            ])
-
-            password = getpass.getpass(cls.indent(password_req_prompt))
-            password_confirm = getpass.getpass(cls.indent("Please confirm your password: "))
-
-            user_auth_handler.set_token_by_registration(email, password, password_confirm, validation_link)
-
+            while True:
```

Comment:
I think if we work on this already, it would be even better (and a considerable convenience boost) if we have this loop around the mail and the password prompt separately s.t. if I mistype my mail I get a warning about this straight away. For this: run the email validator here, too:

```python
email_info = email_validator.validate_email(
                email, test_environment=test_mode
            )
```

and also have a loop to check the passwords are equal.

Reply:
Thanks for the feedback, I have now implemented this and it is definitely a much better user experience. :)
One thing to add: We can also have the case that the entered email already exists. This might not happen very often, but I added an extra request that is sent to the server to also check this directly when the email is entered.

Reply:
yeah that is great, I wondered why you implemented email validation on the server side. good thinking!